processing/src/main/java/io/druid/query/groupby/GroupByQueryConfig.java

```diff
@@ -34,6 +38,26 @@
   @JsonProperty
   private int maxResults = 500000;
 
+  @JsonProperty
+  // Not documented, only used for tests to force spilling
+  private int bufferGrouperMaxSize = Integer.MAX_VALUE;
+
+  @JsonProperty
+  private int bufferGrouperInitialBuckets = -1;
+
+  @JsonProperty
+  // Size of on-heap string dictionary for merging, per-query; when exceeded, partial results will be spilled to disk
+  private long maxMergingDictionarySize = 25 * 1024 * 1024;
```

Comment:
I can add some stuff to groupbyquery.md about how to switch on the new strategy. Will probably call it experimental though.